src/app/machines/views/MachineList/MachineListTable/MachineListTable.tsx

```diff
@@ -143,6 +151,179 @@ const filterColumns = (
   );
 };
 
+const generateRow = ({
+  key,
+  content,
+  hiddenColumns,
+  showActions,
+  classes,
+}: {
+  key: string | number;
+  content: RowContent;
+  hiddenColumns: NonNullable<Props["hiddenColumns"]>;
+  showActions: GenerateRowParams["showActions"];
+  classes: string;
+}) => {
+  const columns = [
+    {
+      "aria-label": columnLabels[MachineColumns.FQDN],
+      key: MachineColumns.FQDN,
+      className: "fqdn-col",
+      content: content[MachineColumns.FQDN],
+    },
+    {
+      "aria-label": columnLabels[MachineColumns.POWER],
+      key: MachineColumns.POWER,
+      className: "power-col",
+      content: content[MachineColumns.POWER],
+    },
+    {
+      "aria-label": columnLabels[MachineColumns.STATUS],
+      key: MachineColumns.STATUS,
+      className: "status-col",
+      content: content[MachineColumns.STATUS],
+    },
+    {
+      "aria-label": columnLabels[MachineColumns.OWNER],
+      key: MachineColumns.OWNER,
+      className: "owner-col",
+      content: content[MachineColumns.OWNER],
+    },
+    {
+      "aria-label": columnLabels[MachineColumns.POOL],
+      key: MachineColumns.POOL,
+      className: "pool-col",
+      content: content[MachineColumns.POOL],
+    },
+    {
+      "aria-label": columnLabels[MachineColumns.ZONE],
+      key: MachineColumns.ZONE,
+      className: "zone-col",
+      content: content[MachineColumns.ZONE],
+    },
+    {
+      "aria-label": columnLabels[MachineColumns.FABRIC],
+      key: MachineColumns.FABRIC,
+      className: "fabric-col",
+      content: content[MachineColumns.FABRIC],
+    },
+    {
+      "aria-label": columnLabels[MachineColumns.CPU],
+      key: MachineColumns.CPU,
+      className: "cores-col",
+      content: content[MachineColumns.CPU],
+    },
+    {
+      "aria-label": columnLabels[MachineColumns.MEMORY],
+      key: MachineColumns.MEMORY,
+      className: "ram-col",
+      content: content[MachineColumns.MEMORY],
+    },
+    {
+      "aria-label": columnLabels[MachineColumns.DISKS],
+      key: MachineColumns.DISKS,
+      className: "disks-col",
+      content: content[MachineColumns.DISKS],
+    },
+    {
+      "aria-label": columnLabels[MachineColumns.STORAGE],
+      key: MachineColumns.STORAGE,
+      className: "storage-col",
+      content: content[MachineColumns.STORAGE],
+    },
+  ];
+
+  return {
+    key,
+    className: classNames(
+      "machine-list__machine",
+      {
+        "truncated-border": showActions,
+      },
+      classes
+    ),
+    columns: filterColumns(columns, hiddenColumns, showActions),
+  };
+};
+
+const generateSkeletonRows = (
+  hiddenColumns: NonNullable<Props["hiddenColumns"]>,
+  showActions: GenerateRowParams["showActions"]
+) => {
+  return Array.from(Array(5)).map((_, i) => {
+    const content = {
+      [MachineColumns.FQDN]: (
+        <DoubleRow
+          primary={<Placeholder>xxxxxxxxx.xxxx</Placeholder>}
```

Comment:
I assume that you decided to use `xxx` is to simulate the length of text this field would normally have, but it results in a confusing label.

Whilst it shouldn't normally be a problem as it's displayed only briefly I'd consider this to be an anti-pattern. Would you consider using a single "loading" aria-label for the table content (or each placeholder) instead?

<img width="1287" alt="Screenshot 2022-08-30 at 14 34 06" src="https://user-images.githubusercontent.com/7452681/187438015-caaff989-7ebb-40e0-a2ce-23e7bb8ec0fd.png">

Reply:
I've added `aria-hidden` hidden to the `<Placeholder>` component (I can't think of a time we'd ever want those elements to be exposed) and added the loading label to the table. Could you double check if that's all how you imagined it? If not I can do a followup to fix it.